### DISCUSSION TITLE
Prioritization between improving syntax and implementing MVP

### DISCUSSION THREAD

**Discussion by bugarela** [informalsystems/quint#210] [User stories]
After discussions started during and after the retreat, and considering Q4 planning is next week, we've stumbled upon a prioritization problem where we are not sure what to do first:

- **Syntax alternatives and user research**: With the goal of finding good syntax constructs that engineers can understand and not be scared of at first glance, our plan is to come up with different syntax (specially for defining transitions/actions) and show them to engineers (at informal and its partners), asking what is their intuition about what a piece of Quint code is doing.
- **MVP**: The first possible MVP we are implementing on Q3 uses the simulator do perform random execution, and it's a good tool for people to try and to learn the language. However, for us to actually start to use the MVP internally, it has to be integrated with model checking. For that, the plan is to generate Apalache IR JSON files from Quint preprocessed specifications, which can be parsed by Apalache and, therefore, model checked.

The work on this model checking integrated MVP is currently planned for the next quarter. After receiving a lot of great feedback on the syntax, we are unsure whether to re-prioritize this and start with syntax-related user research, or to stick with the original plan for the quarter, finish the MVP and leave syntax to be improved next year.

Both tasks are important for us to validate our language usability, in different manners. Writting syntax alternatives can help us validate the surface layer, but people would judge the tool only by looking at code snippets; Finishing the MVP helps us validate the entire workflow from spec writing to model checking, which can lead to more concrete feedback from people actually trying to use the tool for real-world application, but can also be heavily affected by the syntax not being polished. Of course, we want to do **both**, we are just unsure about the order.

Do you have any input that can help us choose what to start with? @tesnimab @milosevic

**Comment by tesnimab** [comment]
Let us prioritize the syntax alternatives and evaluate the time needed to implement the changes.
Implementing the changes would mostly be on the parsing part perhaps; it may not affect the work on the simulator @konnov is doing. 
We have also to discuss whether we should focus on finishing the simulator or starting connecting the model-checker in Q4. For that, we need to do a Q3 retro too.

**Reply by konnov** [reply]
@tesnimab, could you elaborate? Would you like to prioritize "syntax alternatives" over all other work? You are right in that we should discuss the priorities. Also, it is not "either/or". We will prepare the list of work blobs and discuss them at the planning session next week together.

**Comment by konnov** [comment]
Closing this discussion as not being relevant anymore.
